### PR TITLE
Henry/external teaser

### PR DESCRIPTION
**Pull request by chenryhen**
Adds a TeaserIcons component to the structure of a teaser. When certain flags are set, the icons are used to indicate a feature of the teaser. Currently, only `isExternalLink` has an effect.

### CODE REVIEW

File: components/Teaser/icons.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const ATTRIBUTE_ICONS = {
+  isExternalLink: <ExternalLink />
+};
+
+function TeaserIcons({ teaserAttributes }) {
+  return (
+    <div className="Teaser__icons">
+      {teaserAttributes.map(attribute => (
+        <div className="Teaser__icon" key="attribute">
```
Comment (blech75) [inline components/Teaser/icons.js]: right now, if you pass in `hasGallery` and `isVideo`, it appears that the `div.Teaser__icon` gets output. i don't think you want that, right?

also, i think you mean `key={attribute}`.

File: components/Teaser/icons.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          {ATTRIBUTE_ICONS[attribute]}
+        </div>
+      ))}
+    </div>
+  );
+}
+
+TeaserIcons.propTypes = {
+  teaserAttributes: arrayOf(oneOf(['hasGallery', 'isExternalLink', 'isVideo']))
```
Comment (blech75) [inline components/Teaser/icons.js]: mostly a style thing, but i might do something like this:
```js
  teaserAttributes: arrayOf(oneOf(Object.keys(ATTRIBUTE_ICONS))
```
to more explicitly bound the allowed values and mappings together.

File: components/Teaser/icons.spec.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+describe('<TeaserIcons />', () => {
+  describe('sanity check', () => {
+    it('should render', () => {
+      const { wrapper } = setup();
+      expect(wrapper.exists()).toBe(true);
+    });
+  });
+
+  describe('uses attributes properly', () => {
+    it('shows external link icon when `isExternalLink` flag is set', () => {
```
Comment (blech75) [inline components/Teaser/icons.spec.js]: you should mock the icons. something like this should work.
```
jest.mock('../../constants/icons', () => ({
  ExternalLink: jest.fn()
}));
```

File: components/Teaser/index.spec.js [hunk 4]
```diff
@@ -78,6 +78,15 @@ describe('<Teaser />', () => {
           .text()
       ).toEqual(cta);
     });
+
+    it('renders icons when supplied with `props.teaserAttributes`', () => {
+      const teaserAttributes = ['isExternalLink'];
```
Comment (blech75) [inline components/Teaser/index.spec.js]: you should mock `TeaserIcons` in this file.
```js
jest.mock('./icons');
```

File: components/Teaser/icons.js (hunk 1 above)
Comment (chenryhen) [reply components/Teaser/icons.js]: no and yes.

**Pull request merged**